spec/math_spec.rb

```diff
@@ -434,6 +434,50 @@
         expect(n.hermitian?).to be_truthy
       end
     end
+
+    context "#permute_columns for #{dtype}" do
+      it "check that #permute_columns works correctly by considering every premutation of a 3x3 matrix" do
+        n = NMatrix.new([3,3], [1,0,0,
```

Comment:
Can you add a couple tests for matrices larger than 3x3 please?  As I found out in my failed counterexample for your other PR, many of the 3x3 permutations are accidentally the same for the two conventions.

(Doesn't need to be an exhaustive set of permutations of the larger matrix.  Just include a couple that definitely wouldn't be the same.)